recorder.go

```diff
@@ -104,3 +105,72 @@ func (p *Printer) Record(loc *Location, op Operation, stats *Stats) error {
 	p.lastValue = time.Now()
 	return nil
 }
+
+// Full Recorder records all of the evaluations that occur during an optimization
+// run. If Operation is an Evaulation, FullRecorder records the x location and
+// the corresponding field of Location. If Operation is a MajorIteration, or
+// PostIteration, FullRecorder records the full value of the Location struct.
+// Otherwise, FullRecorder records the operation type.
+type FullRecorder struct {
+	Writer io.Writer
+}
+
+func (f FullRecorder) Init() error {
+	return nil
+}
+
+func (f FullRecorder) Record(loc *Location, op Operation, stats *Stats) error {
+	r := RecordLocation{Op: op.String()}
+	switch {
+	case op.isEvaluation():
+		r.Loc.X = loc.X
+		switch op {
+		default:
+			panic("optimize: unknown evaluation operation")
+		case FuncEvaluation:
```

Comment:
Evaluation operations can be ORed together which this switch does not take into account.